src/components/Carousel.tsx

```diff
@@ -1,18 +1,117 @@
-import React from 'react';
+import { Component } from 'react';
 import './Carousel.scss';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li><img src="./img/1.png" alt="1" /></li>
-      <li><img src="./img/1.png" alt="2" /></li>
-      <li><img src="./img/1.png" alt="3" /></li>
-      <li><img src="./img/1.png" alt="4" /></li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type Props = {
+  images: string[];
+  itemWidth: number;
+  frameSize: number;
+  step: number;
+  animationDuration: number;
+  infinity: boolean;
+};
+
+type State = {
+  itemIndex: number,
+};
+
+export class Carousel extends Component<Props, State> {
+  state = {
+    itemIndex: 0,
+  };
+
+  handleButton = (step: number) => () => {
```

Comment:
make more meaningful names
```suggestion
  handleSwipe = (step: number) => () => {
```